File: ios/MonGARSTurboModules.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>
#import "mongars-Swift.h"

@interface RCT_EXTERN_MODULE(LocalLLMModule, RCTEventEmitter)

RCT_EXTERN_METHOD(getAvailableModels:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(downloadModel:(NSString*)modelId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(loadModel:(NSString*)modelId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(deleteModel:(NSString*)modelId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(cancelDownload:(NSString*)modelId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(generateStream:(NSArray*)messages resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(getAvailableSpace:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

@end

File: src/types/MonGARSTurboModules.ts
import type { TurboModule } from 'react-native';
import { Int32, Float } from 'react-native/Libraries/Types/CodegenTypes';
import { CoreMLModel, AIMessage } from './ai';

type NativeAIMessage = Omit<AIMessage, 'id' | 'timestamp' | 'isError'>;

export interface Spec extends TurboModule {
    getAvailableModels(): Promise<CoreMLModel[]>;
    downloadModel(modelId: string): Promise<void>;
    loadModel(modelId: string): Promise<void>;
    deleteModel(modelId: string): Promise<void>;
    cancelDownload(modelId: string): Promise<void>;
    generateText(prompt: string, maxTokens: Int32, temperature: Float, seed: Int32): Promise<string>;
    generateStream(messages: NativeAIMessage[]): Promise<void>;
    stopGeneration(): Promise<void>;
    getAvailableSpace(): Promise<string>;
    addListener(eventName: string): void;
    removeListeners(count: Int32): void;
}


